### PR TITLE
fix: [SPG-1935]: Hotfix for backward compatibility with manager

### PR DESCRIPTION
**Pull request by vcalasansh**
## Describe your changes
Very simple hotfix just to avoid NPE when `DockerArtifactDelegateRequest.shouldFetchDockerV2DigestSHA256` is `null`.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43706)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java [hunk 1]
```diff
@@ -98,9 +98,11 @@ public ArtifactTaskExecutionResponse validateArtifactImage(DockerArtifactDelegat
   }
 
   public ArtifactMetaInfo getArtifactMedataInfo(DockerArtifactDelegateRequest attributesRequest) {
+    boolean shouldFetchDockerV2DigestSHA256 =
+        Boolean.TRUE.equals(attributesRequest.getShouldFetchDockerV2DigestSHA256());
```
Comment (sribalij) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java:102]: Wouldn't this still throw a NPE at `attributesRequest.getShouldFetchDockerV2DigestSHA256()`

**Review by sribalij — COMMENTED** [review]
Let's discuss this offline

**Review by vcalasansh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43706)* status: 2 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. Waiting on team(s): del-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java` line 102 at r2](https://reviewable.io/reviews/harness/harness-core/43706#-NNmC74jS10F1CEp2znS-r2-102:-NNmu8Pi2pCP7OEC64TC:b-baauh7) ([raw file](https://github.com/harness/harness-core/blob/41629f5f557efd4ca46bc0606ac04dad5e7f5833/930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java#L102)):*
<details><summary><i>Previously, sribalij (Sridhar) wrote…</i></summary><blockquote>

Wouldn't this still throw a NPE at `attributesRequest.getShouldFetchDockerV2DigestSHA256()`
</blockquote></details>

Thanks for the comment man! It will not throw NPE here, the NPE was being thrown because `dockerRegistryService.getArtifactMetaInfo` expects `boolean` type and we were passing it as `null` (because `Boolean` can be null, but not `boolean`). Now we are resolving it correctly to a `boolean` type.

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43706)* status: 1. unresolved discussions. Waiting on team(s): del-team.

**Review by sribalij — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43706)* status: Waiting on team(s): del-team. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java` line 102 at r2](https://reviewable.io/reviews/harness/harness-core/43706#-NNmC74jS10F1CEp2znS-r2-102:-NNmxVvu47bLph0yssvI:bdjrg1z) ([raw file](https://github.com/harness/harness-core/blob/41629f5f557efd4ca46bc0606ac04dad5e7f5833/930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/docker/DockerArtifactTaskHandler.java#L102)):*
<details><summary><i>Previously, vcalasansh (Vinícius Calasans) wrote…</i></summary><blockquote>

Thanks for the comment man! It will not throw NPE here, the NPE was being thrown because `dockerRegistryService.getArtifactMetaInfo` expects `boolean` type and we were passing it as `null` (because `Boolean` can be null, but not `boolean`). Now we are resolving it correctly to a `boolean` type. 
</blockquote></details>

That makes sense. Thanks!

**Pull request merged**